TestCases/strayalien/0068-feel-equality/0068-feel-equality.dmn

```diff
@@ -0,0 +1,748 @@
+        <allowedAnswers>false (boolean)</allowedAnswers>
+        <variable typeRef="boolean" name="deep_003"/>
+        <literalExpression>
+            <text>[1,2,[3, 4]] = [1,2,[4, 3]]</text>
+        </literalExpression>
+    </decision>
+
+    <decision name="deep_004" id="_deep_004">
+        <description>Tests FEEL expression: '[1,2,{a: [3,4]}] = [1,2,{a: [3,4], b: "foo"}]' and expects result: 'false (boolean)'</description>
+        <question>Result of FEEL expression '[1,2,{a: [3,4]}] = [1,2,{a: [3,4], b: "foo"}]'?</question>
+        <allowedAnswers>false (boolean)</allowedAnswers>
+        <variable typeRef="boolean" name="deep_004"/>
+        <literalExpression>
+            <text>[1,2,{a: [3,4]}] = [1,2,{a: [3,4], b: "foo"}]</text>
+        </literalExpression>
+    </decision>
+
+    <decision name="deep_005" id="_deep_005">
+        <description>Tests FEEL expression: '{a: {b: "foo", c: [1,2]}} = {a: {b: "foo", c: [1,2]}}' and expects result: 'true (boolean)'</description>
+        <question>Result of FEEL expression '{a: {b: "foo", c: [1,2]}} = {a: {b: "foo", c: [1,2]}}'?</question>
+        <allowedAnswers>true (boolean)</allowedAnswers>
+        <variable typeRef="boolean" name="deep_005"/>
+        <literalExpression>
+            <text>{a: {b: "foo", c: [1,2]}} = {a: {b: "foo", c: [1,2]}}</text>
+        </literalExpression>
+    </decision>
+
+    <decision name="deep_006" id="_deep_006">
+        <description>Tests FEEL expression: '{a: {c: [1,2], b: "foo"}} = {a: {b: "foo", c: [1,2]}}' and expects result: 'true (boolean)'</description>
+        <question>Result of FEEL expression '{a: {c: [1,2], b: "foo"}} = {a: {b: "foo", c: [1,2]}}'?</question>
+        <allowedAnswers>true (boolean)</allowedAnswers>
+        <variable typeRef="boolean" name="deep_006"/>
+        <literalExpression>
+            <text>{a: {c: "bar", b: "foo"}} = {a: {b: "foo", c: "bar"}}</text>
+        </literalExpression>
+    </decision>
+
+    <decision name="deep_007" id="_deep_007">
+        <description>Tests FEEL expression: '{a: {b: "foo", c: [1,2]}} = {a: {b: "foo", c: [2,1]}}' and expects result: 'false (boolean)'</description>
+        <question>Result of FEEL expression '{a: {b: "foo", c: [1,2]}} = {a: {b: "foo", c: [2,1]}}'?</question>
+        <allowedAnswers>false (boolean)</allowedAnswers>
+        <variable typeRef="boolean" name="deep_007"/>
+        <literalExpression>
+            <text>{a: {b: "foo", c: [1,2]}} = {a: {b: "foo", c: [2,1]}}</text>
+        </literalExpression>
+    </decision>
+
+    <decision name="singleton_001" id="_singleton_001">
+        <description>Tests FEEL expression: '[100] = 100' and expects result: 'false (boolean)'</description>
+        <question>Result of FEEL expression '[100] = 100'?</question>
+        <allowedAnswers>false (boolean)</allowedAnswers>
+        <variable typeRef="boolean" name="singleton_001"/>
+        <literalExpression>
+            <text>[100] = 100</text>
```

Comment:
The spec says:

> For convenience of notation, a singleton list L, when used in an expression where a list is not expected, behaves as if L[1] is written.

it follows on with filtering, function invocation examples, but is not exhaustive in other example like the one proposed here.

Just based on the former definition, one could argue the other operand is a number (`100`), hence a singleton list of a number (`[100]`) would fall in the former definition; hence by "_behaving as if L[1] is written_" would lead to this expression to be actually true, while the test case is asserting it to be false.